stix2/properties.py

```diff
@@ -17,82 +23,109 @@ class Property(object):
           value, and
         - a `default()` function that returns the fixed value.
         (Default: `None`)
 
     Subclasses can also define the following functions.
 
     - `def clean(self, value) -> any:`
-        - Transform `value` into a valid value for this property. This should
-          raise a ValueError if such no such transformation is possible.
-    - `def validate(self, value) -> any:`
-        - check that `value` is valid for this property. This should return
-          a valid value (possibly modified) for this property, or raise a
-          ValueError if the value is not valid.
-          (Default: if `clean` is defined, it will attempt to call `clean` and
-          return the result or pass on a ValueError that `clean` raises. If
-          `clean` is not defined, this will return `value` unmodified).
+        - Return a value that is valid for this property. If `value` is not
+          valid for this property, this will attempt to transform it first. If
+          `value` is not valid and no such transformation is possible, it should
+          raise a ValueError.
     - `def default(self):`
         - provide a default value for this property.
         - `default()` can return the special value `NOW` to use the current
             time. This is useful when several timestamps in the same object need
             to use the same default value, so calling now() for each field--
             likely several microseconds apart-- does not work.
 
-    Subclasses can instead provide lambda functions for `clean`, and `default`
-    as keyword arguments. `validate` should not be provided as a lambda since
-    lambdas cannot raise their own exceptions.
+    Subclasses can instead provide a lambda function for `default as a keyword
+    argument. `clean` should not be provided as a lambda since lambdas cannot
+    raise their own exceptions.
     """
 
-    def _default_validate(self, value):
+    def _default_clean(self, value):
         if value != self._fixed_value:
             raise ValueError("must equal '{0}'.".format(self._fixed_value))
         return value
 
-    def __init__(self, required=False, fixed=None, clean=None, default=None, type=None):
+    def __init__(self, required=False, fixed=None, default=None, type=None):
         self.required = required
         self.type = type
         if fixed:
             self._fixed_value = fixed
-            self.validate = self._default_validate
+            self.clean = self._default_clean
             self.default = lambda: fixed
-        if clean:
-            self.clean = clean
         if default:
             self.default = default
 
     def clean(self, value):
-        raise NotImplementedError
-
-    def validate(self, value):
-        try:
-            value = self.clean(value)
-        except NotImplementedError:
-            pass
         return value
 
+    def __call__(self, value=None):
+        if value is not None:
+            return value
+
 
```

Comment:
All code paths in \_\_call\_\_ should return a value, or none should.  If value is None, execution falls off the end without a return (which means None is returned, but that should be explicit).  As written so far, this method is a no-op anyway.  It's the same as the one-liner `return value`.